### PR TITLE
add mlr3spatial

### PR DESCRIPTION
**Pull request by be-marc**

### CODE REVIEW

File: content/post/2021-12-01-mlr3spatial/index.Rmd [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+categories:
+  - R
+tags:
+  - mlr3spatial
+  - CRAN
+---
+
+We are happy to announce that [mlr3spatial](https://mlr3spatial.mlr-org.com/) has been released on [CRAN](https://cran.r-project.org/web/packages/mlr3spatial/index.html). 
+[mlr3spatial](https://mlr3spatial.mlr-org.com/) simplifies the handling of spatial objects in the [mlr3](https://github.com/mlr-org) ecosystem.
+Before [mlr3spatial](https://mlr3spatial.mlr-org.com/), the user had to extract raw values from spatial objects to train a model or predict a map.
```
Comment (pat-s) [inline content/post/2021-12-01-mlr3spatial/index.Rmd]: "predict a map" 

-> We are not really predicting a map, we're creating a map from the predicted values which also inherit spatial information.
Also we might need to explain what "raw values" are here.

File: content/post/2021-12-01-mlr3spatial/index.Rmd [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+date: '2021-12-01'
+slug: mlr3spatial
+categories:
+  - R
+tags:
+  - mlr3spatial
+  - CRAN
+---
+
+We are happy to announce that [mlr3spatial](https://mlr3spatial.mlr-org.com/) has been released on [CRAN](https://cran.r-project.org/web/packages/mlr3spatial/index.html).
```
Comment (pat-s) [inline content/post/2021-12-01-mlr3spatial/index.Rmd]: We could add the release date :)

File: content/post/2021-12-01-mlr3spatial/index.Rmd [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+tags:
+  - mlr3spatial
+  - CRAN
+---
+
+We are happy to announce that [mlr3spatial](https://mlr3spatial.mlr-org.com/) has been released on [CRAN](https://cran.r-project.org/web/packages/mlr3spatial/index.html). 
+[mlr3spatial](https://mlr3spatial.mlr-org.com/) simplifies the handling of spatial objects in the [mlr3](https://github.com/mlr-org) ecosystem.
+Before [mlr3spatial](https://mlr3spatial.mlr-org.com/), the user had to extract raw values from spatial objects to train a model or predict a map.
+
+Now, mlr3 [tasks](https://mlr3.mlr-org.com/reference/Task.html) can directly read from spatial objects via specialized [data backends](https://mlr3.mlr-org.com/reference/DataBackend.html).
```
Comment (pat-s) [inline content/post/2021-12-01-mlr3spatial/index.Rmd]: List the backend names explicitly?

File: content/post/2021-12-01-mlr3spatial/index.Rmd [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+Such tasks can be used to train a model or to perform resampling just like any other mlr3 task.
+We support spatial raster objects created by the packages [terra](https://CRAN.R-project.org/package=terra), [raster](https://CRAN.R-project.org/package=raster) and [stars](https://CRAN.R-project.org/package=stars).
```
Comment (pat-s) [inline content/post/2021-12-01-mlr3spatial/index.Rmd]:
```suggestion
We support spatial raster objects created by the [terra](https://CRAN.R-project.org/package=terra), [raster](https://CRAN.R-project.org/package=raster) and [stars](https://CRAN.R-project.org/package=stars) packages.
```

File: content/post/2021-12-01-mlr3spatial/index.Rmd [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+Additionally, vector data is handled with the package [sf](https://CRAN.R-project.org/package=sf).
+
+The `predict_raster()` function makes a map with predictions from an mlr3 learner.
```
Comment (pat-s) [inline content/post/2021-12-01-mlr3spatial/index.Rmd]: We should mention that we use `geom_sf()` under the hood and briefly mention what is happening behind the scenes (i.e. what steps we take away from the user).

PS: "making a map" should be rephrased :)

File: content/post/2021-12-01-mlr3spatial/index.Rmd [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+We only have to pass a task with a spatial data backend which provides the data and spatial reference.
+To avoid memory issues with large raster files, the prediction is done in chunks.
+One after the other, the raster chunks are loaded into memory and the prediction is written to disk.
+Finally, the complete map is available on disk. 
+The learner can also make use of future-based parallelization to accelerate the predictions.
```
Comment (pat-s) [inline content/post/2021-12-01-mlr3spatial/index.Rmd]: We might want to explain in more detail how the chunking works and how the user can control it (-> argument) and what the default is doing.

**Review by pat-s — CHANGES_REQUESTED** [review]
Thanks!

Blog posts provide the venue to include figures and code - we should use it, even when just showing a basic example here 🙂️
This post is about a new package, so we should use the opportunity to write a bit more.

We can make use of the vignette contents and their examples.
I think the post could be (should be) at least twice as long :)

This is the final/important piece to get the message out to the users and get people to use the package :)

File: content/post/2021-12-01-mlr3spatial/index.Rmd (hunk 3 above)
Comment (be-marc) [reply content/post/2021-12-01-mlr3spatial/index.Rmd]: We would just repeat line 19 and 20.

File: content/post/2021-12-01-mlr3spatial/index.Rmd (hunk 5 above)
Comment (be-marc) [reply content/post/2021-12-01-mlr3spatial/index.Rmd]: Converting an sf object to data.table?

File: content/post/2021-12-01-mlr3spatial/index.Rmd (hunk 6 above)
Comment (be-marc) [reply content/post/2021-12-01-mlr3spatial/index.Rmd]: The user can adapt the chunks size (`chunksize`) to the available memory.
The default chunk size is 200 Megabytes which should work on most consumer consumer grade machines.
If more memory is available, a larger chunk size accelerates the prediction process. 

?

File: content/post/2021-12-01-mlr3spatial/index.Rmd (hunk 1 above)
Comment (be-marc) [reply content/post/2021-12-01-mlr3spatial/index.Rmd]: > Also we might need to explain what "raw values" are here.

The user had to create tabular data from spatial objects to train a model or predict a map.

?

> We are not really predicting a map, we're creating a map from the predicted values which also inherit spatial information.

Sure that's technically correct but it's much simpler to just say map. Everybody knows  what a map is. But you are right. I change it in the other sentences.

File: content/post/2021-12-01-mlr3spatial/index.Rmd (hunk 3 above)
Comment (pat-s) [reply content/post/2021-12-01-mlr3spatial/index.Rmd]: The backend names are `DataBackendRaster` and `DataBackendVector`. I don't see these referenced in l.19 and 20.

File: content/post/2021-12-01-mlr3spatial/index.Rmd (hunk 5 above)
Comment (pat-s) [reply content/post/2021-12-01-mlr3spatial/index.Rmd]: `predict_raster()` does way more, doesn't it?

File: content/post/2021-12-01-mlr3spatial/index.Rmd (hunk 1 above)
Comment (pat-s) [reply content/post/2021-12-01-mlr3spatial/index.Rmd]: > The user had to create tabular data from spatial objects to train a model or predict a map.

Sounds better 👍 

> Sure that's technically correct but it's much simpler to just say map. Everybody knows what a map is. But you are right. I change it in the other sentences.

I know this sounds a bit nit-picky but keep in mind the audience: not only spatial people will read it and even if, being on point with terminology helps to convince them :)

File: content/post/2021-12-01-mlr3spatial/index.Rmd (hunk 6 above)
Comment (pat-s) [reply content/post/2021-12-01-mlr3spatial/index.Rmd]: I think the term "chunk size" needs to be explained/introduced in more detail. 
Also we should mention that this behavior might change if {terra} changes something WRT to memory usage etc. 

We should also explain we have chosen 200 MB as most consumer machines have more available memory. 

> If more memory is available, a larger chunk size accelerates the prediction process.

Yes, we should add that this needs to be enforced by the user.

File: content/post/2021-12-01-mlr3spatial/index.Rmd [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+stack = stars::read_stars(tif)
+
+backend = as_data_backend(stack)
+task = as_task_regr(backend, target = "layer.1")
+
+print(task)
+```
+
+For large raster files with millions of values it helps to predict in parallel.
+To enable this, set `learner$parallel_predict = TRUE` and initiate a parallel plan via {future}.
```
Comment (pat-s) [inline content/post/2021-12-01-mlr3spatial/index.Rmd]:
```suggestion
To enable this, set `learner$parallel_predict = TRUE` and initiate a parallel plan via {future}, e.g. via `plan("multisession")`.
```

**Comment by pat-s** [comment]
- Can we name the code chunks for better navigation?
- I'll have a look again on the post after all comments have been resolved :)

**Review by pat-s — APPROVED** [review]
@be-marc I am now fine with the post, if you're ok with my changes then we can publish :)

**Pull request merged**